### ISSUE TITLE
Incorrect loss function

### ISSUE THREAD

**Issue by angmc** [mattsherar/Temporal_Fusion_Transform#2]
I believe the loss function is incorrect.
 Currently calculated:

errors = target[:, i] - preds[:, i]
losses.append(torch.max((q-1) * errors,   q * errors ).unsqueeze(1))

however the google github has the loss calculated in the following:
prediction_underflow = y - y_pred
q_loss = quantile * tf.maximum(prediction_underflow, 0.) + (1. - quantile) * tf.maximum(-prediction_underflow, 0.)

The correct calculation should be something like this:

losses.append(
q * torch.max(errors, torch.zeros_like(errors)) +  (1. - q) * torch.max(-errors, torch.zeros_like(errors)  )
